src/librustc/ty/relate.rs

```diff
@@ -584,7 +584,40 @@ pub fn super_relate_consts<R: TypeRelation<'tcx>>(
         // FIXME(const_generics): we should either handle `Scalar::Ptr` or add a comment
         // saying that we're not handling it intentionally.
 
-        // FIXME(const_generics): handle `ConstValue::ByRef` and `ConstValue::Slice`.
+        (
+            ConstValue::Slice { data: alloc_a, start: offset_a, end: end_a },
+            ConstValue::Slice { data: alloc_b, start: offset_b, end: end_b },
+        ) => {
+            let len_a = end_a - offset_a;
```

Comment:
This code effectively already exists at https://github.com/rust-lang/rust/blob/7e3fb8aea12ead737544db95a3d6c93988105b89/src/librustc_mir/hair/pattern/mod.rs#L1528-L1552, so maybe you could abstract this common functionality out of both?

Reply:
I added a function to `rustc::mir:interpret::value` for getting the bytes out of a slice const value.